examples/phase_estimator.py

```diff
@@ -0,0 +1,182 @@
+
+        """A quantum circuit with the following recursive structure.
+        ---H----R_2---R_3---R_4--------------------------------------
+                 |     |     |
+        ---------@-----|-----|-----H----R_2---R_3--------------------
+                       |     |           |     |                       = QFT_inv
+        ---------------@-----|-----------@-----|-----H----R_2--------
+                             |                 |           |
+        ---------------------@-----------------@-----------@-----H---
+
+        where R_k is an inverse phase shift U as below:
+        U|0> = |0>
+        U|1> = exp(-2 Pi i / 2^k)|1>
+
+        The number of qubits can be arbitrary.
+        """
+
+        qubits = list(qubits)
+        while len(qubits) > 0:
+            q_head = qubits.pop(0)
+            yield cirq.H(q_head)
+            for i, qubit in enumerate(qubits):
+                yield self.cr_inverse(i+2)(qubit, q_head)
+
+
+def run_estimate(phi, qnum, repeats):
+
+    """Construct the following phase estimator circuit and execute simulations.
+                                     ---------
+    ---H---------------------@------|         |---[m3]---
+                             |      |         |
+    ---H---------------@-----|------|         |---[m2]---
+                       |     |      | QFT_inv |
+    ---H---------@-----|-----|------|         |---[m1]---
+                 |     |     |      |         |
+    ---H---@-----|-----|-----|------|         |---[m0]---
+           |     |     |     |       ---------
+    -------U----U^2---U^4---U^8--------------------------
+
+    The measurement results are translated to the estimated phase with the
+    formula: phi = m0*(1/2) + m1*(1/2)^2 + m2*(1/2)^3 + m3*(1/2)^4 + ...
+    """
+
+    qubits = [None] * qnum
+    for i in range(len(qubits)):
+        qubits[i] = cirq.GridQubit(0, i)
+    ansilla = cirq.GridQubit(0, len(qubits))
+
+    ops = [cirq.H(q) for q in qubits]
+    ops += [ControlledGate(phase_op((2**i)*phi))(qubits[qnum-i-1], ansilla)
+            for i in range(qnum)]
+    ops.append(QftInverse()(*qubits))
+    ops += [cirq.measure(qubits[len(qubits)-i-1], key='m{}'.format(i))
+            for i in range(qnum)]
+    circuit = cirq.Circuit.from_ops(*ops)
+    simulator = cirq.google.XmonSimulator()
+    result = simulator.run(circuit, repetitions=repeats)
+    return result
+
+
+def experiment(qnum, repeats=100):
+
+    """Execute the phase estimator cirquit with multiple settings and
+    show results.
+    """
+
+    print('Estimation with {}qbits.'.format(qnum))
+    print('Actual, Estimation (Raw binary)')
+    errors = []
+    for phi in np.arange(0, 1, 0.1):
+        result = run_estimate(phi, qnum, repeats)
+        measurements_bool = zip(*[result.measurements['m{}'.format(i)].flatten()
+                                  for i in range(qnum)])
+        measurements_bin = [('{:d}'*qnum).format(*m) for m in measurements_bool]
```

Comment:
Does `TrialResult.histogram` cover this use case?

```python
counter = result.histogram(fold_func=lambda ms: ...)
```

Reply:
fixed.